core/domain/question_services_test.py

```diff
@@ -30,91 +29,101 @@ class QuestionServicesUnitTest(test_utils.GenericTestBase):
     def setUp(self):
         """Before each individual test, create dummy user."""
         super(QuestionServicesUnitTest, self).setUp()
-
-        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
-        user_services.create_new_user(self.owner_id, self.OWNER_EMAIL)
-        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.signup(self.TOPIC_MANAGER_EMAIL, self.TOPIC_MANAGER_USERNAME)
+        self.signup(self.NEW_USER_EMAIL, self.NEW_USER_USERNAME)
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.topic_manager_id = self.get_user_id_from_email(
+            self.TOPIC_MANAGER_EMAIL)
+        self.new_user_id = self.get_user_id_from_email(
+            self.NEW_USER_EMAIL)
+        self.editor_id = self.get_user_id_from_email(
+            self.EDITOR_EMAIL)
+
+        self.set_admins([self.ADMIN_USERNAME])
+        self.set_topic_managers([self.TOPIC_MANAGER_USERNAME])
+
+        self.topic_manager = user_services.UserActionsInfo(
+            self.topic_manager_id)
+        self.admin = user_services.UserActionsInfo(self.admin_id)
+        self.new_user = user_services.UserActionsInfo(self.new_user_id)
+        self.editor = user_services.UserActionsInfo(self.editor_id)
+
+        self.question_id = question_services.get_new_question_id()
+        self.question = self.save_new_question(
+            self.question_id, self.editor_id,
+            self._create_valid_question_data('ABC'))
 
     def test_get_question_by_id(self):
```

Comment:
You should also test the case where "strict" is not provided. And also the error cases for both.

Reply:
Done.